soc/riscv/riscv-privilege/nuclei/gd32vf103/Kconfig.defconfig.series

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_RISCV_GIGADEVICE_GD32VF103
+
+source "soc/riscv/riscv-privilege/nuclei/gd32vf103/Kconfig.defconfig.gd32vf103*"
+
+config KERNEL_ENTRY
+	default "__nuclei_start"
+
+config SOC_SERIES
+	default "nuclei"
```

Comment:
I' ll apply it.
The change make folder structure changes
soc/riscv/riscv-privilege/nuclei/gd32vf103 to soc/riscv/riscv-privilege/gd32vf103.
(nuclei intermediate folder is removed.)

From the convention in soc/riscv/riscv-privilege/, gigadevice_gd32vf103 may be better. 

The SoC folder is usually named [manufacturerer]_[product] and
Set SOC_SERIES to the same value as the folder name.
(Used in soc/riscv/riscv-privilege/CMakeLists.txt.)

Reply:
Hold on, I miss that, sorry ! Better keep as you already proposed.
We need find another way to solve this. Maybe it will be better define GD32_ variables at modules/hal_gigadevice.

Reply:
OK. keep original PR.

Reply:
Accepting @gmarull comment, 
https://github.com/zephyrproject-rtos/zephyr/pull/34970#pullrequestreview-819199212
I changes SOC_SERIES to gd32vf103.
Please see additional comment at the bottom of this PR.